droidlet/perception/semantic_parsing/tests/test_nsp_loading.py

```diff
@@ -0,0 +1,31 @@
+import os
+import unittest
+
+from droidlet.perception.semantic_parsing.nsp_transformer_model.query_model import NSPBertModel as Model
+
+NLU_MODEL_DIR = os.path.join(
+    os.path.dirname(__file__), "../../../../agents/craftassist/models/semantic_parser/"
+)
+NLU_DATA_DIR = os.path.join(
+    os.path.dirname(__file__), "../../../../agents/craftassist/datasets/annotated_data/"
+)
+
+class TestNSPModel(unittest.TestCase):
+    def setUp(self):
+        nsp_data_dir = NLU_DATA_DIR
+        nsp_model_dir = os.path.join(NLU_MODEL_DIR, "ttad_bert_updated")
+        self.model = Model(model_dir=nsp_model_dir, data_dir=nsp_data_dir)
+
+    def test_model_parse(self):
+        chat = 'come here'
+        logical_form = self.model.parse(chat=chat)
+        self.assertEqual(type(logical_form), dict)
```

Comment:
Couple of thoughts here:
1. Is there something more than a type check we can do here? 
2. Can you also add a test for a case when model loading throws an exception - for eg when model_dir is wrong / empty etc.

Reply:
Yea, model testing is tricky. I thought just type check might not be sufficient because it would pass even when an empty dict is returned. If an empty dict means that the NSP is broken in some way, then we should add a stronger check (since we wouldn't want this test to pass then). If not, then this is fine.

Reply:
Yeah I added a stricter check to all three : `self.assertTrue("dialogue_type" in logical_form)` since we expect "dialogue_type" key in the dict at the minimum